Small tidy in formatting of generated text

forejtv opened the pull request:
Two small improvements in generated junit code.

1) Do not print "initialise test parameters comment" if there are no parameters. Closes diffblue/verification-engine-utils#343.
2) Print out fewer empty lines. This required various changes as newline characters were added quite generously :-). The main change for this was in d0d5f7e, where handling of new lines and indentation had to be changed in order to make sure that empty string is returned by get_mock_prelude when no real code was added. Closes diffblue/verification-engine-utils#342.

peterschrammel commented on src/test_gen/java_test_source_factory.cpp:
```diff
@@ -565,16 +562,20 @@ bool reference_factoryt::add_func_call_parameters(
-  std::set<irep_idt> already_done;
-  for(const auto& symbol : needed)
+
+  if (needed.size() > 0)
```
no space after if; no spaces around >

peterschrammel commented on src/test_gen/java_test_source_factory.cpp:
```diff
@@ -565,16 +562,20 @@ bool reference_factoryt::add_func_call_parameters(
-      continue;
-    indent(result, 2u);
-    add_declare_and_assign(result,st,symbol,inputs[symbol.name],true);
+    indent(result,2u) += "/* initialize test parameters */\n";
```
no space around +=

peterschrammel commented on src/test_gen/java_test_source_factory.cpp:
```diff
@@ -565,16 +562,20 @@ bool reference_factoryt::add_func_call_parameters(
+      if(!already_done.insert(symbol.name).second)
+        continue;
+      indent(result, 2u);
+      add_declare_and_assign(result,st,symbol,inputs[symbol.name],true);
```
spaces after comma

peterschrammel commented on src/test_gen/java_test_source_factory.cpp:
```diff
@@ -1139,11 +1140,18 @@ std::string generate_java_test_case_from_inputs(const symbol_tablet &st, const i
     indent(post_mock_setup_result)+=force_instantiate(desc.classname);
     post_mock_setup_result+=";\n";
   } 
+  std::string mock_prelude = ref_factory.mockenv_builder.get_mock_prelude();
```
no spaces around =

peterschrammel commented on src/test_gen/java_test_source_factory.cpp:
```diff
@@ -1139,11 +1140,18 @@ std::string generate_java_test_case_from_inputs(const symbol_tablet &st, const i
   } 
+  std::string mock_prelude = ref_factory.mockenv_builder.get_mock_prelude();
+  if (!mock_prelude.empty())
+    result += ref_factory.mockenv_builder.get_mock_prelude() + "\n";
```
no spaces etc in the following lines

peterschrammel commented on src/test_gen/java_test_source_factory.cpp:
```diff
@@ -1139,11 +1140,18 @@ std::string generate_java_test_case_from_inputs(const symbol_tablet &st, const i
+
   // Finalise happens here because add_func_call_parameters et al
   // may have generated mock objects.
   std::string mock_final = ref_factory.mockenv_builder.finalise_instance_calls();
```
finalize

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -20,10 +20,9 @@ Function: mock_envrionment_builder
 mock_environment_builder::mock_environment_builder(unsigned int ip)
```
unsigned int -> unsigned

forejtv replied:
I am not sure I follow. There seem to be quite a few occurrences of "unsigned int" in the codebase (>350). If we want to change them, maybe it would make sense not to do it as part of this PR.

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -46,10 +45,10 @@ std::string mock_environment_builder::register_mock_instance(
-		 << prelude_newline;
-  return prelude_newline+instanceList+".add("+instancename+");";
+    mock_prelude << prelude_indent
+                 << "java.util.ArrayList<" << tyname << "> " << instanceList
```
instanceList -> instance_list

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -46,10 +45,10 @@ std::string mock_environment_builder::register_mock_instance(
+    mock_prelude << prelude_indent
+                 << "java.util.ArrayList<" << tyname << "> " << instanceList
+		 << "=new java.util.ArrayList<" << tyname << ">();\n";
+  return prelude_indent+instanceList+".add("+instancename+");\n";
```
instance_name

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -46,10 +45,10 @@ std::string mock_environment_builder::register_mock_instance(
-  return prelude_newline+instanceList+".add("+instancename+");";
+    mock_prelude << prelude_indent
+                 << "java.util.ArrayList<" << tyname << "> " << instanceList
+		 << "=new java.util.ArrayList<" << tyname << ">();\n";
```
type_name?

forejtv replied:
I am not familiar enough with the code to say if this is the case or not.

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -86,9 +85,10 @@ void mock_environment_builder::constructor_call(
                                          {targetclass,false},retval,true);
```
spaces after commas

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -207,22 +207,23 @@ method_answer* mock_environment_builder::add_to_answer_list(
-		 << prelude_newline
+    mock_prelude << prelude_indent
+                 << "final java.util.ArrayList<" << boxed_type << "> " << al
+		 << "=new java.util.ArrayList<" << boxed_type << ">();\n"
```
not enough indent

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -338,12 +337,12 @@ std::string mock_environment_builder::finalise_instance_calls()
-    result << "for(" << cname << " " << instanceIter << " : " << instanceList
-	   << ')' << prelude_newline << "  ";
+    result << prelude_indent
+           << "for(" << cname << " " << instanceIter << " : " << instanceList
```
instance_iter, instance_list

peterschrammel commented on src/test_gen/mock_environment_builder.cpp:
```diff
@@ -450,18 +451,16 @@ void mock_environment_builder::print_statements(
       break;
     case init_statement::STATEMENT:
-      printto << s.statementText << ';' << prelude_newline;
+      printto << prelude_indent << s.statementText << ";\n";
```
statement_text

forejtv commented:
@peterschrammel I have fixed what you pointed out and also some other bits that were against guidelines. But I noticed that fixing everything would be quite major work, so I would like to not do a thorough style change (unless of course it's in code that I introduced in this PR).
